src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -736,6 +739,31 @@ class RandomDateGenerator : public Generator<bsoncxx::types::b_date> {
     const std::chrono::milliseconds _max;
 };
 
+
+class IncGenerator : public Generator<int64_t> {
+public:
+    IncGenerator(const Node& node, GeneratorArgs generatorArgs)
+        : _actorId{generatorArgs.actorId},
+          _step{extract(node, "step", "^Inc").to<int64_t>()},
+          _multiplier{extract(node, "multiplier", "^Inc").to<int64_t>()},
+          _start{extract(node, "start", "^Inc").to<int64_t>()}
+          {_counter=_start+_actorId*_multiplier;}
+
+    int64_t evaluate() override {
+        auto inc_value = _counter;
+        _counter += _step;
+        return inc_value;
+    }
+
+private:
+    int64_t _actorId;
+    int64_t _counter;
+    int64_t _step;
+    int64_t _multiplier;
+    int64_t _start;
```

Comment:
looks like you don't actually need to store `_actorId`, `_multiplier`, or `_start`, you just use those to compute the initial value of `_counter`.

Also you generally want to initialize things in the same order as they are declared in the struct/class (`: _actorId{...}, _counter{...}` etc). We're loose with this though.